customtransforms/custom_single_json_retriever.py

```diff
@@ -0,0 +1,52 @@
+from transformer import Transformer
+from utilities import flat
+
+
+class SingleJsonRetriever(Transformer):
+    '''
+    Transforms perfherder data into the standardized data format.
+    '''
+    entry_number = 0
+
+    def transform(self, data):
+        ret = []
+        self.entry_number += 1
+
+        flatted_data = flat(data, [])
+
+        for k, v in flatted_data.items():
+            ret.append({
+                'data': v,
+                'xaxis': self.entry_number,
+                'subtest': k
+            })
+
+        return ret
+
+    def merge(self, data):

```

Comment:
This function should be moved into the `Transformer` base class with the name `merge_subtests` and then you can call that function from this one to do the work. Let's prevent duplication as much as possible. Otherwise we will have to eventually do a larger refactor which is a waste of time if we could have prevented it from the start.

Reply:
Can you explain why you have a merge function which doesn't do anything?